### PR TITLE
Cross section

### PR DESCRIPTION
**Pull request by NadHobeika**
Implemented DSM according to meeting feedback on 6-5-2020.

### CODE REVIEW

File: code/groundTin.py [hunk 1]
```diff
@@ -169,6 +172,32 @@ def interpolate_triangle(self, tr, pt):
...
 
+    def interpolate_edge(self, edge_0, edge_1, pt):
+        """
+        Explanation:
+        Find the interpolated value on an edge
+        ---------------
+        Input:
+        edge_0: first endpoint coordinates
+        edge_1: second endpoint coordinates
+        pt: point on the edge [x,y]
```
Comment (Constantijn-Dinklo) [inline code/groundTin.py]: Make these like this:
name : type - explination (so pt : [x,y] - point on the edge)
If [x,y] is a predefined type (like integer) just type int.
Right now we have no idea what type/format the edges are in.

File: code/groundTin.py [hunk 2]
```diff
@@ -207,17 +236,166 @@ def create_cross_section(self, edge_list, source, receiver, source_tr, receiver_
...
+        Adds buildings to the DTM cross section
+        ---------------
+        Input:
+        dtm_vertices: List of vertices of cross section
+        dtm_edges: list of edges, each edge is a tuple of 2 vertices and tuple of associated building polygons.[(__,__), (__, ..._)
+        Buildings: list of shapely polygons with height attribute (how will this be exactly?)
+        ---------------
+        Output:
+        a list with vertices in the cross section plane
+        a list of edges which use the generated vertices.
```
Comment (Constantijn-Dinklo) [inline code/groundTin.py]: Change format here too.

File: code/groundTin.py [hunk 3]
```diff
@@ -207,17 +236,166 @@ def create_cross_section(self, edge_list, source, receiver, source_tr, receiver_
...
+        """
+        cross_section_vertices = []
+        cross_section_edges = []
+        for edge in dtm_edges:
+            shapely_line = LineString([dtm_vertices[edge[0][0]], dtm_vertices[edge[0][1]]])
+            count = len(cross_section_vertices)
+            if not edge[1]:
+                if dtm_vertices[edge[0][0]] not in cross_section_vertices:
```
Comment (Constantijn-Dinklo) [inline code/groundTin.py:271]: Is an edge of the format:
[((x1,y1),(x2,y2)), (b1, b2, ..., bx), ..... , .... ]
So ((x1,y1),(x2,y2)), (b1, b2, ..., bx) is the format of 1 edge?

And edge[0][0] would get (x1,y1)?

**Review by Constantijn-Dinklo — CHANGES_REQUESTED** [review]
I've made some comments of the changes we need.

File: code/groundTin.py (hunk 3 above)
Comment (NadHobeika) [reply code/groundTin.py:271]: Yes, exactly.

Comment (NadHobeika) [reply code/groundTin.py:271]: Actually edge[0][0] would get (v0, v1) where v is the index of the vertex

**Review by Constantijn-Dinklo — APPROVED** [review]
Changes are good.

**Pull request merged**